### ISSUE TITLE
Error if add space before "http" in linked template uri

### ISSUE THREAD

**Issue by StephenWeatherford** [microsoft/vscode-azurearmtools#1212]
Action: onTemplateGraphAvailable
Error type: Error
Error Message: [UriError]: Scheme contains illegal characters.

Version: 0.15.0-alpha2
OS: darwin
OS Release: 19.6.0
Product: Visual Studio Code - Insiders
Product Version: 1.54.0-insider
Language: en

<details>
<summary>Call Stack</summary>

```
D extensionHostProcess.js:56:42271
new f extensionHostProcess.js:56:43204
new m extensionHostProcess.js:56:44834
Function.parse extensionHostProcess.js:56:43963
Object.getParameterDefinitionsFromLinkedTemplate getParameterDefinitionsFromLinkedTemplate.js:11:78
LinkedTemplateScope.assignLinkedFileReferences templateScopes.js:350:86
Object.assignTemplateGraphToDeploymentTemplate linkedTemplates.js:144:27
AzureRMTools.js:1542:43
Object.callWithTelemetryAndErrorHandlingSync vscode-azureextensionui/callWithTelemetryAndErrorHandling.js:53:16
AzureRMTools.onTemplateGraphAvailable AzureRMTools.js:1528:35
p.fire extensionHostProcess.js:55:1856
startArmLanguageServer.js:315:46
startArmLanguageServer.js:12:71
startArmLanguageServer.js:8:12
startArmLanguageServer.js:314:105
Object.callWithTelemetryAndErrorHandlingSync vscode-azureextensionui/callWithTelemetryAndErrorHandling.js:53:16
onNotifyTemplateGraph startArmLanguageServer.js:314:31
startArmLanguageServer.js:211:21
startArmLanguageServer.js:12:71
startArmLanguageServer.js:8:12
startArmLanguageServer.js:210:96
handleNotification vscode-jsonrpc/main.js:489:43
processMessageQueue vscode-jsonrpc/main.js:260:17
Immediate._onImmediate vscode-jsonrpc/main.js:247:13
processImmediate timers.js:456:21
process.callbackTrampoline async_hooks.js:120:14
```

</details>

**Comment by StephenWeatherford** [comment]
Also:

  "templateLink": {
                    "uri": "file:hello",
                    "contentVersion": "1.0.0.0" 
                },

**Comment by StephenWeatherford** [comment]
And worse:

"uri": "http:>"

**Comment by StephenWeatherford** [comment]
'  httzp:'

**Comment by StephenWeatherford** [comment]
Fixed by #1217
